Feature/kan 25 backend jwt authentication

RezaBG opened the pull request.

IshanRastogi commented:
The `branch name` doesn't match with the `work done` in the branch. Are you sure that you've completed the tasks that you wanted to ?? 

According to the branch name: you should be doing some `backend jwt authentication` here, however, from the looks of it the only thing accomplished here is the fixing of your `TaskStatus` and associated `case` related issues.

RezaBG commented:
- TaskStatus and Case Handling: Addressed and fixed issues related to TaskStatus and case normalization for task statuses.
- Endpoint Testing: Successfully tested all 14 endpoints, ensuring they behave as expected.

IshanRastogi commented on tests/test_insert_task.py:
```diff
@@ -0,0 +1,14 @@
+from sqlalchemy.orm import Session
+from src.models import Task, TaskStatus
+
+def test_insert_task(db: Session):
```
can you run this individual task and check the result ?

IshanRastogi reviewed (approved):
Merge this branch and then let's debug the next bit.
